File: arrays/stonks.js
function stonks(wsPrices){
    let min_i = 0;
    let profit = 0;
    
    for(let i = 1;i<wsPrices.length;i++){

        if(wsPrices[i] >= wsPrices[min_i]){
            let possibleProfit = wsPrices[i] - wsPrices[min_i];
            
            if(possibleProfit > profit){
                profit = wsPrices[i] - wsPrices[min_i];
            }

        }

        if(wsPrices[i] < wsPrices[min_i]){
            min_i = i;
        }

    }
    return profit;
}

let wsPrices = [7,1,5,3,6,4];
console.log(stonks(wsPrices));

File: arrays/stonks.py
def stonks(wsPrices):
    min_i = 0
    profit = 0

    for i in range(len(wsPrices)):
        if(wsPrices[i] >= wsPrices[min_i]):
            possibleProfit = wsPrices[i] - wsPrices[min_i]

            if(possibleProfit > profit):
                profit = possibleProfit
        
        if wsPrices[i] < wsPrices[min_i]:
            min_i = i
    
    return profit


wsPrices = [7,1,5,3,6,4]
print (stonks(wsPrices))

